airflow/www/views.py

```diff
@@ -654,9 +654,10 @@ def index(self):
             dags_query = session.query(DagModel).filter(~DagModel.is_subdag, DagModel.is_active)
 
             if arg_search_query:
+                escaped_arg_search_query = arg_search_query.replace("_", "\\_")
```

Comment:
nit: feel free to iginore
```suggestion
                escaped_arg_search_query = arg_search_query.replace("_", r"\_")
```